_posts/2024-04-25-integrating-backups-into-gitops.md

```diff
@@ -0,0 +1,322 @@
+---
+author: michaelcade
+date: 2024-04-25 16:21:37 +0100
+description: ""
+featured: false
+image: ""
+image_caption: ""
+layout: post
+published: true
+tags: [kasten, gitops, cicd, argocd, pipelines]
+title: "Integrating Backups into your GitOps Pipeline "
+---
+
+# Integrating Backups into your GitOps Pipeline 
+
+This post is going to uncover some of the realities that we face around companies adopting a GitOps model within their infrastructure and operational procedures. Ultimately GitOps is going to rely on Git as a source control system. 
+
+The industry definition can be found below: 
+
+*GitOps is an operational framework based on DevOps practices, like continuous integration/continuous delivery (CI/CD) and version control, which automates infrastructure and manages software deployment.*
+
+[Atlassian - Is GitOps the next big thing in DevOps?](https://www.atlassian.com/git/tutorials/gitops)
+
+## My thoughts on GitOps 
+
+I may also refer to GitOps in this post according to CI/CD as well because this can be seen as maybe the predecessor to GitOps. My take is that maybe you are not a software development house today so maybe CI "Continious Integration" is not something that crosses your mind, however CD (Continous Delivery) will absolutely be something every single customer running some sort of software should be considering deploying their software the GitOps way. 
+
+![](/images/posts/2024-04-25-integrating-backups-into-gitops/Picture1.png)
+
+In looking at the diagram above if you are consuming off the shelf software within your environments, as software companies are speeding up delivery of software where v1 to v2 is now weeks vs a year or even longer, then GitOps is going to help you stay in control of these updates and provide you a good way to keep track but also provide a rollback workflow when bad things happen. Now if you only have one off the shelf software in your environment maybe this is a stretch but if you have lots of COTS (Commercial off the shelf software) then a GitOps way is going to help with that control. 
+
+## Introducing ArgoCD 
+
+In our example we are going to focus on ArgoCD and specifically Kubernetes as our platform to run our applications. 
+
+*Argo CD is a Kubernetes-native continuous deployment (CD) tool. Unlike external CD tools that only enable push-based deployments, Argo CD can pull updated code from Git repositories and deploy it directly to Kubernetes resources.*
```

Comment:
not sure I understand what you mean exactly. Are you saying that there is a direct relationship between git and kubernetes with Argo but other CD tools can not do that ? If yes I'm not sure I agree.